helix-core/src/main/java/org/apache/helix/task/WorkflowContext.java

```diff
@@ -48,22 +48,27 @@
     LAST_PURGE_TIME,
     StartTime, // TODO this should be named JOB_SCHEDULED_START_TIME, it's not the actual start time of the job
     NAME
-    }
+  }
 
   public static final int NOT_STARTED = -1;
   public static final int UNFINISHED = -1;
 
+  private boolean isModified;
```

Comment:
Somewhere in the class, you need to note that `isModified` need to be updated every time there's a change. Future developers may forget and break this.

Reply:
Thanks for the coment. Done.